src/main/java/bitflow4j/steps/database/ElasticsearchUtil.java

```diff
@@ -113,6 +122,72 @@ private IndexRequest generateIndexRequest(Sample sample) throws IOException {
+        requestIndex.indices(indexName);
+        boolean indexExists = client.indices().exists(requestIndex, RequestOptions.DEFAULT);
+        logger.log(Level.INFO, String.format("indexExists: %s", indexExists));
+        if (!indexExists) {
+            CreateIndexRequest requestCreate = new CreateIndexRequest(indexName);
+            //requestCreate.
+
+            XContentBuilder builder = XContentFactory.jsonBuilder();
+            builder.startObject();
+            // Brackets only for readability
+            {
+                builder.startObject("_doc");
+                {
+                    builder.startObject("properties");
+                    {
+                        // Create timestamp field as date which is saved in milliseconds
+                        builder.startObject("timestamp");
+                        {
+                            builder.field("type", "date");
+                            builder.field("index", "true");
+                            builder.field("format", "epoch_millis");
+                        }
+                        builder.endObject();
+
+                        // Create identifierKey field as keyword
+                        builder.startObject(identifierKey);
+                        {
+                            builder.field("type", "keyword");
+                        }
+                        builder.endObject();
+
+                        // Create
+                        String[] header = sample.getHeader().header;
+                        for (String metric : header) {
+                            builder.startObject(metric);
+                            {
+                                builder.field("type", "float");
+                            }
+                            builder.endObject();
+                        }
+                    }
+                    builder.endObject();
+                }
+                builder.endObject();
+            }
+            builder.endObject();
+            requestCreate.mapping("_doc", builder);
+            try {
+                CreateIndexResponse createIndexResponse = client.indices().create(requestCreate, RequestOptions.DEFAULT);
+                boolean acknowledged = createIndexResponse.isAcknowledged();
+                boolean shardsAcknowledged = createIndexResponse.isShardsAcknowledged();
+                logger.log(Level.INFO, String.format("Created index for indexName '%s' and got response '%s' and '%s'.", indexName, acknowledged, shardsAcknowledged));
+            }
+            catch (ElasticsearchStatusException elasticEx){
+                //Can happen if multiple forks try to create this index, just ignore the message that it already was added
+                if (!elasticEx.getMessage().contains("resource_already_exists_exception")){
+                    throw elasticEx;
+                }
+            }
+        }
+        indexCheckDone = true;
```

Comment:
Please move this assignment into the if clause in generateIndexRequest(), to have the check and assignment together.